extensions/amp-story-360/0.1/amp-story-360.js

```diff
@@ -385,33 +381,39 @@ export class AmpStory360 extends AMP.BaseElement {
   }
 
   /**
-   * Creates a device orientation listener and sets gyroscopeControls_ state.
-   * If listener is not called in 1000ms, remove listener and resume animation.
-   * This happens on desktop browsers that support deviceorientation but aren't in motion.
+   * Listens for deviceorientation events.
+   *
+   * Some browsers support the 'deviceorientation' event but never call it.
+   * This waits for one call before initiating a constant listener.
    * @private
    */
   enableGyroscope_() {
-    this.gyroscopeControls_ = true;
-    this.togglePermissionClass_(true);
-
-    const checkNoMotion = this.timer_.delay(() => {
-      this.gyroscopeControls_ = false;
-      if (this.isReady_ && this.isPlaying_) {
-        this.animate_();
-      }
-    }, 1000);
-
-    let rafTimeout;
-
-    this.win.addEventListener('deviceorientation', (e) => {
-      if (this.isReady_ && this.isOnActivePage_) {
-        // Debounce onDeviceOrientation_ to rAF.
-        rafTimeout && this.win.cancelAnimationFrame(rafTimeout);
-        rafTimeout = this.win.requestAnimationFrame(() =>
-          this.onDeviceOrientation_(e)
-        );
+    // Listen for one call before initiating.
+    listenOncePromise(this.win, 'deviceorientation').then(() => {
+      this.gyroscopeControls_ = true;
+      this.togglePermissionClass_(true);
+      // Debounce onDeviceOrientation_ to rAF.
+      let rafTimeout;
+      this.win.addEventListener('deviceorientation', (e) => {
+        if (this.isReady_ && this.isOnActivePage_) {
+          rafTimeout && this.win.cancelAnimationFrame(rafTimeout);
+          rafTimeout = this.win.requestAnimationFrame(() =>
+            this.onDeviceOrientation_(e)
+          );
+        }
+      });
+      // Display discovery animation.
+      if (this.isOnActivePage_) {
```

Comment:
About when this animation is triggered:
- Do we want to trigger it on each 360 media of a Story?
- Do we re-trigger it when restarting the page (navigation forward then backward)?
- If I understand correctly the animation plays on `layoutCallback` or upon gyroscope permission grant. Can we change it so it's triggered when the page becomes active?

Reply:
+1 and also: can we have a way to disable it ? the stories I'm working on will have text right where the animation runs, it'll likely not look good. 

Besides, having an attribute to  toggle that animation might also allow creators to only show it on eg the first page that has a 360 player and turn it off on subsequent pages. Because basing that only on permission grant would probably not work as expected on androd.

Reply:
We can file a new ticket for a way to opt-out, I'd like PM/UX to contribute to the conversation.

Reply:
This really feels like part of the story education system, would be interesting to align requirements with that

Reply:
An option to swap the position of the animation would work as well. Or have it vertically centered with a darker overlay that would dim out the text layer in the content no matter its location.

Reply:
Thanks for this great feedback :) 

> Do we want to trigger it on each 360 media of a Story?
> Do we re-trigger it when restarting the page (navigation forward then backward)?

If we assume the user will recognize the feature once they are aware of it, we can show it on the first active 360 page while the feature is enabled.

> Can we have a way to disable it?

If we consider this optional we shouldn’t have it at all. 
Anything in between creates an inconsistent UI experience.

> This really feels like part of the story education system, would be interesting to align requirements with that.

This is component level rather than story level so I think it’s okay for it to be visually tailored to the component. 
I agree that it would be good to be sure this is aligned with our education experiences

> An option to swap the position of the animation would work as well. Or have it vertically centered with a darker overlay that would dim out the text layer in the content no matter its location.

Definitely  needs  contrast against content behind it. The layout and look of it needs to be consistent. 
It’s important to see some of the background clearly so the user sees the content moving.

cc @hongcatlover

Reply:
SGTM!

For the record, I wasn't referring to the UX re: aligning with story education, but more how we keep track of whether or not to show this to users.  We should hook into the education system's capabilities around (a) determining whether the user has seen this before and/or should see it (again), and (b) allowing different players to have different configurations for when/how it should be shown.

Reply:
That SGTM as well, please ping us when you get a chance to write the code!